### DISCUSSION TITLE
how to append a component tag using jquery

### DISCUSSION THREAD

**Discussion by haripriya-goswami** [adonisjs/core#2923] [Help] [answered]
I want to append a component template in the existing edge template using jquery after receiving data from ajax request. How is it possible? or is there any other way to do it?

```
$("div").append(`@!component('components/profile', { profile: ${newProfile} })`)
```
This code appends it as a string instead of a tag

**Comment by RomainLanz** [comment, marked as answer]
Hey @ir4h! 👋 

You are mixing backend and frontend code.

The best for you would be to generate the HTML in the server and return it so you can append it in the client.

```ts
// Server code...
return view.render('components/profile', { profile })
```

```ts
// frontend code
const html = await fetch('...')

$("div").append(html)
```

**Reply by haripriya-goswami** [reply]
Hey, thanks a lot. It's working now! :)
